docs/content/guides/dagster-pipes/subprocess/reference.mdx

```diff
@@ -0,0 +1,242 @@
+---
+title: "Modifying existing code to work with Dagster Pipes | Dagster Docs"
+description: "This page shows ways to execute external code with Dagster Pipes with different entities in the Dagster systen."
+---
+
+# Reference
+
+This reference shows usage of Dagster Pipes with other entities in the Dagster system. For a step-by-step walkthrough, refer to the [Dagster Pipes tutorial](/guides/dagster-pipes/subprocess).
+
+## Specifying environment variables and extras
+
+When launching the subprocess, you may want to make environment variables or additoional parameters available in the external process. Extras can be arbitrary, user-defined parameters made available on the context object in the external process.
+
+<TabGroup>
+<TabItem name="Provide in the Dagster code">
+
+The `run` method to the `PipesSubprocessClient` resource also accepts `env` and `extras` , which allow you to specify environment variables and extra arguments when executing the subprocess:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/extras_env_dagster_code.py
+import os
+import shutil
+
+from dagster import (
+    AssetExecutionContext,
+    PipesSubprocessClient,
+    asset,
+    file_relative_path,
+)
+
+
+@asset
+def subprocess_asset(
+    context: AssetExecutionContext, pipes_subprocess_client: PipesSubprocessClient
+):
+    cmd = [shutil.which("python"), file_relative_path(__file__, "external_code.py")]
+    return pipes_subprocess_client.run(
+        command=cmd,
+        context=context,
+        extras={"foo": "bar"},
+        env={
+            "MY_ENV_VAR_IN_SUBPROCESS": os.environ["MY_ENV_VAR"],
```

Comment:
Is there a reason we wouldn't use `EnvVar` here on the Dagster side?

Reply:
just thought it might be too much to explain with `EnvVar` being passed in asset function inputs. i could be convinced tho.

Reply:
Do we need to explain it? Or is linking to documentation sufficient?

We've been telling users that `EnvVar` is best practice, so I feel like our examples should reflect that unless there's a good reason not to. If we don't feel like just linking here is fine, we could say something like:

"We're using `os.environ` in this example, but Dagster's recommendation is to use `EnvVar` in production." 

Or something along those lines - wdyt?

Reply:
further simplified the example to not access the current env var values. so we don't need to worry about conflicting recommendations.